src/dawn/Optimizer/PassSetBlockSize.cpp

```diff
@@ -0,0 +1,72 @@
+//===--------------------------------------------------------------------------------*- C++ -*-===//
+//                          _
+//                         | |
+//                       __| | __ ___      ___ ___
+//                      / _` |/ _` \ \ /\ / / '_  |
+//                     | (_| | (_| |\ V  V /| | | |
+//                      \__,_|\__,_| \_/\_/ |_| |_| - Compiler Toolchain
+//
+//
+//  This file is distributed under the MIT License (MIT).
+//  See LICENSE.txt for details.
+//
+//===------------------------------------------------------------------------------------------===//
+
+#include "dawn/Optimizer/PassSetBlockSize.h"
+#include "dawn/IIR/IIRNodeIterator.h"
+#include "dawn/Optimizer/OptimizerContext.h"
+
+namespace dawn {
+
+PassSetBlockSize::PassSetBlockSize() : Pass("PassSetBlockSize") {
+  dependencies_.push_back("PassSetStageName");
+}
+
+bool PassSetBlockSize::run(const std::shared_ptr<iir::StencilInstantiation>& stencilInstantiation) {
+  OptimizerContext* context = stencilInstantiation->getOptimizerContext();
+
+  const auto& IIR = stencilInstantiation->getIIR();
+
+  std::array<unsigned int, 3> blockSize{0, 0, 0};
+  if(!context->getOptions().block_size.empty()) {
+    std::string blockSizeStr = context->getOptions().block_size;
+    std::istringstream idomain_size(blockSizeStr);
+    std::string arg;
```

Comment:
can we check that the argument is properly formatted here?

Reply:
stoi is throwing if conversion is not possible

Reply:
stoi is throwing if conversion is not possible

Reply:
so we only need to check that the string is empy after the last getline

Reply:
ok